docs(cloning): add brief and hld draft (#551)
Vizzuality/marxan-cloud: docs/download-upload-clone/brief.md

# Cloning Marxan Cloud projects and scenarios

> This document describes the current high-level overview of the Marxan Cloud platform user features related to cloning of projects of scenarios. It is a working document, meant to gather feedback on the requirements before moving on to high level design and detailed implementation plan.

Reviewer: ```suggestion
This document describes the high-level design of the Marxan Cloud platform user features related to cloning of projects or scenarios.
```

## Aim

Users should be able to clone Marxan Cloud projects for a range of purposes:
- A user may see a community project (i.e. those that can be browsed without 
logging in to the platform) and wish to create a copy within their own account, to study it and modify their own copy
- A student being trained on Marxan may need to clone an existing project (whether public, from the community section, or a private project to which they have been given access by the project owner), in order to study the project and learn how to set up and run Marxan on the project
- A user may need to download one of their projects, in order to archive a copy that can be later imported into a different Marxan Cloud instance
- A user may need to set up a local instance of the open source Marxan Cloud platform, importing into it a copy of one or more projects previously downloaded from the flagship Marxan Cloud instance

Users should likewise be able to clone Marxan Cloud scenarios (within the same project):

> - Users may wish to undertake a new analysis by changing parameters of an existing scenario, while keeping the original (cloned) scenario intact, so that they can later

Reviewer: ```suggestion
- Users may wish to undertake a new analysis by changing parameters of an existing scenario, while keeping the original (cloned) scenario intact, so that they can later compare the two scenarios.
```

# User actions

> In all these use cases, the core operation being described is that of cloning a project or a scenario: this may be done within the platform itself by clicking on a “Clone project” or “Clone scenario” button, or (in the case of projects) as a two-step process, first downloading an archive file containing all the data needed to successively upload the same project to the same or another Marxan Cloud instance, and secondly actually importing the archive.

Reviewer 2: Is it contemplated that the user can change the name of the cloned project/scenario in order to save it?
